cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.api.schedule.TriggeringScheduleInfo;
+import co.cask.cdap.app.store.Store;
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import co.cask.cdap.internal.app.runtime.ProgramOptionConstants;
+import co.cask.cdap.internal.app.store.RunRecordMeta;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import co.cask.cdap.proto.ops.DashboardProgramRunRecord;
+import co.cask.http.HttpResponder;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * {@link co.cask.http.HttpHandler} to handle program run operation dashboard and reports for v3 REST APIs
+ */
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3)
+public class OperationsDashboardHttpHandler extends AbstractAppFabricHttpHandler {
+  private static final Gson GSON = new Gson();
+  private final Store store;
+
+  @Inject
+  public OperationsDashboardHttpHandler(Store store) {
+    this.store = store;
+  }
+
+  // TODO: [CDAP-13351] Need to support getting scheduled program runs if start + duration is ahead of the current time
+  @GET
+  @Path("/dashboard")
+  public void readDashboardDetail(FullHttpRequest request, HttpResponder responder,
+                                  @QueryParam("start") long start,
+                                  @QueryParam("duration") int duration,
```

Comment:
will be good to name the variables  with unit `startTimeSecs` and `durationSecs`

Reply:
according to the design, you would throw 400 exception on invalid time range, non existent namespace, etc. however those doesn't seem to be handled. is the design doc up to date ?